### ISSUE TITLE
Large Number Warning When used in Anonymous Function

### ISSUE THREAD

**Issue by jeremyjh** [rrrene/credo#435]
### Precheck
No open bug.
### Environment

* Credo version (`mix credo -v`): 0.8.6
* Erlang/Elixir version (`elixir -v`): 20.0 / 1.5.0
* Operating system: MacOS Sierra 10.12.6

### What were you trying to do?
`mix credo --strict` on a file with contents:

```Elixir
defmodule Demo.LargeNumberAnonWarning do
  @moduledoc false

  def harmless_function do
    say_num = fn num ->
      IO.puts "The number is: #{inspect num}"
    end

    say_num.(10_000)
  end
end
```

### Expected outcome

No warnings.

### Actual outcome

"Large Numbers Should be Written With Underscores"
``` shell
mix credo --strict
Checking 1 source file ...

  Code Readability
┃
┃ [R] ↗ Large numbers should be written with underscores: 10_000
┃       lib/large_numbers_anon.ex:9:15 #(Demo.LargeNumberAnonWarning.harmless_function)
```

The issue only appears when the large number is passed to a variable binding an anonymous function.

**Comment by rrrene** [comment]
That's a bug. Thx for reporting it! :+1:

**Comment by philipdexter** [comment]
The values returned from `:elixir_tokenizer.tokenize` are incorrect. See the following for an example

```elixir
iex(1)> :elixir_tokenizer.tokenize('a.(1_000)', 1, [])                         
{:ok, 1, 11,                                                                   
 [{:identifier, {1, 1, 2}, :a}, {:dot_call_op, {1, 2, 3}, :.},
  {:"(", {1, 4, 5}}, {:number, {1, 5, 10}, 1000}, {:")", {1, 10, 11}}]}
```

Perhaps https://github.com/elixir-lang/elixir/issues/4907 and https://github.com/elixir-lang/elixir/pull/4910 are relevant.

**Comment by rrrene** [comment]
Thanks for reporting this 😀  It is now fixed on `master`.

You can try this by setting the Credo dep to

```elixir
{:credo, github: "rrrene/credo"}
```

Please report back if your issue is solved! 👍

**Comment by jeremyjh** [comment]
@rrrene I confirmed this resolves the issue, thanks.
